Targets transition mega PR (#2)
USGS-R/ds-pipelines-targets-2-course: responses/00-overview.md

**Welcome to the second installment of "introduction to data pipelines"** at USGS, @{{ user.username }}!! :sparkles:

> We're assuming you were able to navigate through the [intro-to-pipelines](https://lab.github.com/USGS-R/intro-to-pipelines) course and that you learned a few things about organizing your code for readability, re-use, and collaboration. You were also introduced to two key things through the makefile: a way to program connections between functions and files, and the concept of a dependency manager that skips parts of the workflow that don't need to be re-run. 

Reviewer: Does this link need to change now that each targets course has a different repo (and name?) from the original ds-pipelines courses?

Author: Good catch - I might make a second issue to update all of the links to the other courses because I don't yet know what they will be. Is it based on this line? https://github.com/USGS-R/ds-pipelines-1-course/blob/master/config.yml#L1

Reviewer: Seems likely that it's that field that matters, yes, but I haven't found the docs to confirm or refute.

Author: captured in #5 for now and will check once I have the courses up for testing

> - Data science work should be organized thoughtfully. As Jenny Bryan notes, "File organization and naming are powerful weapons against chaos".

Reviewer: and may need a new link in line 8, too

Author: captured in #5 for now since I don't know what the URL will be

> - Capture all of the critical phases of project work with descriptive directories and function names, including how you "got" the data (in practice, we often use `fetch` for this phase).

Reviewer: ```suggestion
- Capture all of the critical phases of project work with descriptive directories and function names, including how you "got" the data.
```
The 'fetch' thing just seems like too much detail, diluting the "capture all critical phases" part

> - Turn your scripts into a collection of functions, and modify your thinking to connect deliberate outputs from these functions ("targets") to generate your final product.

Reviewer: "deliberate" outputs? I assume this language is meant to distinguish from [unintentional] side effects, but I doubt that's clear to learners.

Author: Would changing to "connect single outputs" make this idea clearer? Or just deleting "deliberate"?

Reviewer: Yeah, just delete "deliberate", I think

- "Skip the work you don't need" by taking advantage of a dependency manager. There was a video that covered a bit of `make`, and you were asked to experiment with `targets`.

> - Investing in _efficient_ reproducibility helps projects scale up with confidence. 

Reviewer: ```suggestion
- Invest in _efficient_ reproducibility to scale up projects with confidence. 
```

> This last concept was not addressed directly but we hope that the small exercise of seeing rebuilds in action got you thinking about projects that might have much more lengthly steps (e.g., several downloads or geo-processing tasks that take hours instead of seconds).

Reviewer: ```suggestion
This last concept was not addressed directly, but we hope that the small exercise of seeing rebuilds in action got you thinking about projects that might have much more lengthly steps (e.g., several downloads or geo-processing tasks that take hours instead of seconds).
```
